Stat Box Updates

anechol opened the pull request:
## Description
<!-- REQUIRED: add a short description of this update -->
Fixes Stat Box styling to be more in line with Figma mockup. Data attribute has been updated to type "String."

[Stat Box in Figma](https://www.figma.com/file/9Km09NjlZHYWsMP7EGT8tI/%5BWIP%5D-Sage-3-%E2%80%94-Admin-Components?node-id=3435%3A5)

## Screenshots
<!-- OPTIONAL(recommended): Show any visual updates -->
|  Before  |  After  |
|--------|--------|
<img width="460" alt="statbox" src="https://user-images.githubusercontent.com/14791307/118686509-ab965e00-b7c9-11eb-9323-cdc2461aae96.png">|<img width="428" alt="Screen Shot 2021-05-18 at 3 36 16 PM" src="https://user-images.githubusercontent.com/14791307/118720149-de9f1880-b7ee-11eb-8be9-7a6b699cfffb.png">

## Testing in `sage-lib`

### Rails
- View [Rails component](http://localhost:4000/pages/component/stat_box)
- Check that component UI exists and nothing is broken

### React
- View [Storybook](http://localhost:4100/?path=/docs/sage-statbox--default)
- Check that component UI exists and nothing is broken

## Testing in `kajabi-products`
<!-- REQUIRED: Provide general notes describing this change in order for QA to verify the changes within `kajabi-products`. Follow this format: Describe this PR, its impact level (LOW/MEDIUM/HIGH/BREAKING), and where it can be tested. If this a new feature on existing component, indicate places you can demonstrate it has not had adverse effects.
  Read more here: https://github.com/Kajabi/sage-lib/wiki/Version-Bump-Process
  IMPORTANT: Once merged, the list below should be transferred to the anticipated version bump PR -->
(LOW) Updates styling and data attribute. No effect on existing work in Kajabi Products.

## Related
<!-- OPTIONAL: link to related issues or PRs for context -->
[MAN-1385](https://kajabi.atlassian.net/browse/MAN-1385)

philschanely commented on docs/lib/sage_rails/app/views/sage_components/_sage_stat_box.html.erb:
```diff
@@ -1,13 +1,14 @@
-elsif change_type == "down"
-  label_configs = { color: "danger", value: change_value, icon: "caret-down" } 
+if change_type == "positive"
+  label_configs = { color: "published", value: change_value, icon: change_icon } 
```
Here is the hope to allow for an icon override? Seems we may still want to provide preset icons in the conditionals below like:

`... icon: (change_icon ? change_icon : "caret-up") ...` etc.

philschanely reviewed (commented):
Great style changes! Some comments below regarding icon setup but otherwise ready to go. LMK if you'd like to discuss more in a call.

anechol commented:
Per brainstorming session with @philschanely, I've removed the option for a custom icon and reverted back to defaults/fallbacks. If a custom scenario or icon is required for the label, developers have the option to pass in whatever override they feel is appropriate.

monicawheeler commented:
Should the "After" screenshot show an arrow on the percent increase?

anechol commented:
Oops! Good catch @monicawheeler. Updated with new image!

monkeypox8 commented on docs/lib/sage_rails/app/views/sage_components/_sage_stat_box.html.erb:
```diff
@@ -20,11 +20,13 @@ end
-    <% if component.timeframe.present? %>
-      <span class="sage-stat-box__timeframe"><%= component.timeframe %></span>
-    <% end %>
+  <main class="sage-stat-box__body sage-grid-template-te">
```
Hmm, I may be mistaken but is it acceptable to use multiple `main` elements on a page and/or as a descendant of `article`? At least per the current W3C spec this has been disallowed: https://www.w3.org/TR/html52/grouping-content.html#the-main-element, but I recall it being a point of contention between the W3C and WHATWG in the past

anechol replied:
Oh, thanks so much for catching this! I don't know why I thought I read a `<main>` element as an acceptable child of `<article>`. Looks like this isn't the case. I've changed `<main>` to `<div>`.

monkeypox8 reviewed (approved):
👍🏼 🔥  LGTM, great work!

philschanely reviewed (approved):
Looking good!
